spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -35,26 +35,32 @@
 class Param:
     # The constructor
     def __init__(self):
-        self.shift_AP = 32  # 0#32  # shift the centerline towards the spine (in voxel).
-        self.size_AP = 11  # 41#11  # window size in AP direction (=y) (in voxel)
-        self.size_RL = 1  # 1 # window size in RL direction (=x) (in voxel)
+        self.shift_AP = 32  # shift the centerline towards the spine (in voxel).
+        self.size_AP = 11  # window size in AP direction (=y) (in voxel)
+        self.size_RL = 1  # window size in RL direction (=x) (in voxel)
         self.size_IS = 19  # window size in IS direction (=z) (in voxel)
-        self.shift_AP_visu = 15  # 0#15  # shift AP for displaying disc values
-        self.smooth_factor = [3, 1, 1]  # [3, 1, 1]
-        self.gaussian_std = 1.0  # STD of the Gaussian function, centered at the most rostral point of the image, and used to weight C2-C3 disk location finding towards the rostral portion of the FOV. Values to set between 0.1 (strong weighting) and 999 (no weighting).
+        self.shift_AP_visu = 15  # shift AP for displaying disc values
+        self.smooth_factor = [3, 1, 1]
         self.path_qc = None
 
     # update constructor with user's parameters
     def update(self, param_user):
-        list_objects = param_user.split(',')
-        for object in list_objects:
-            if len(object) < 2:
+        for param in param_user.split(','):
+            try:
+                key, value = param.split('=', maxsplit=1)
+                value = int(value)
+            except ValueError:
                 printv('ERROR: Wrong usage.', 1, type='error')
-            obj = object.split('=')
-            if obj[0] == 'gaussian_std':
-                setattr(self, obj[0], float(obj[1]))
+                sys.exit(1)
+            if key in ['shift_AP', 'size_AP', 'size_RL', 'size_IS', 'shift_AP_visu']:
+                setattr(self, key, value)
+            elif key == 'gaussian_std':
+                # TODO: remove 'gaussian_std' completely for release 6.0
+                printv('WARNING: gaussian_std parameter is currently ignored, '
+                       'and will be removed in a later version.', 1, type='warning')
```

Comment:
Ah, one other comment. I would love to maybe come up with a procedure for returning to deprecation notices like these?

Maybe like, a specific syntax we could include in the comment, like:


```suggestion
                 # DEP: remove 'gaussian_std' completely two releases after v5.5
                printv('WARNING: gaussian_std parameter is currently ignored, '
                       'and will be removed in a later version.', 1, type='warning')
```

Then we could include instructions in the "preparing for a new release" dev Wiki page to do a project-wide search for `# DEP:` a week or two prior to the release due-date.

Reply:
I'm in favour of using a standardized syntax, yes. How about `# TODO(v6.0): remove gaussian_std completely`? I think `TODO` is already a widely recognized string, and the project doesn't seem to contain `# TODO(` as a substring yet.

From [our versioning guidelines](https://intranet.neuro.polymtl.ca/geek-tips/contributing.html#versioning-a-href-versioning-id-versioning-a), it sounds like we're doing something like [semantic versioning](https://semver.org/spec/v2.0.0.html), so I assumed the deprecation would last until the next major release, v6.0 in this case. Then again, I don't know if we have any sort of timeline for when v6.0 happens. Would you instead prefer v5.7 (so the deprecation warning appears during v5.6), or maybe v5.8 (so the warning appears during v5.6 and v5.7)?

(As an aside, I tend to prefer deadlines (for example, "by v5.7") instead of timeouts ("two releases after v5.5"), as expressed in [this little blog post](https://guxi.me/posts/timeout-vs-deadline/).)

Reply:
> From [our versioning guidelines](https://intranet.neuro.polymtl.ca/geek-tips/contributing.html#versioning-a-href-versioning-id-versioning-a), it sounds like we're doing something like [semantic versioning](https://semver.org/spec/v2.0.0.html), so I assumed the deprecation would last until the next major release, v6.0 in this case.

So... SCT's versioning is a bit tricky at the moment. Before I was hired, we were indeed following semantic versioning! But, IMO there were four main issues with that:

1. SCT's dev team is small, and we're not currently fast/organized enough to plan out both bugfix/patch releases and feature releases. (At least, not without some changes to my/our workflow?)
2. SCT's feature development is more sporadic, and often dependent on whether or not we have dedicated grad students developing new image processing approaches. So, it can be hard(er) to manage/schedule patch vs. feature releases?
3. SCT is not pip installable, which means:
   - Our userbase is has ended up composed of primarily non-developer folks, who are less likely to have semantic expectations for version numbers.
   - There isn't an easy way to update SCT beyond wiping the installation folder entirely, then reinstalling from scratch. (This makes patch releases... functionally useless? Since installing a patch release is equally as laborious as installing, say, a minor release.) If SCT _was_ pip installable, being able to upgrade via pip to the newest patch version would make semantic versioning (and more frequent releases via PyPI) a lot more appealing, I think?
4. Our users tend to stick with a specific version for conducting a study. i.e. they're not likely to stay on top of new versions as rapidly as typical consumers of Python packages. (You'll often see bug reports or questions on the forum which start with "I'm using SCT 4.X..." because that's the version they used a year or two ago in their last study.) 

Historically, all of these factors combined resulted in SCT only ever releasing once there were significant enough changes to _warrant_ a new release. So, you'd see 0-1 patch releases for every minor release -- basically nonexistent. 

During a meeting last year, I asked "Does it even make sense to follow semantic versioning anymore?" with the end result being us ditching semantic versioning to stick entirely to minor/major releases. 

(But, even with minor/major releases only, it's unclear to me what change would be dramatic enough to warrant a 6.0 release. If anything, our release numbering is more of a marketing tool at the moment than a system that conveys developer intent? I expect that once something radical happens (pip installable SCT, windows release, etc.) then that's when we'll choose 6.0, as a marker for a new era of SCT, aha.)

Still, if you have thoughts or opinions about this, I think our approach to versioning definitely worth revisiting. And, at the very least, there should be a dev wiki section documenting the methodology here. 😅

Reply:
> (But, even with minor/major releases only, it's unclear to me what change would be dramatic enough to warrant a 6.0 release. If anything, our release numbering is more of a marketing tool at the moment than a system that conveys developer intent? I expect that once something radical happens (pip installable SCT, windows release, etc.) then that's when we'll choose 6.0, as a marker for a new era of SCT, aha.)

Given what I've written here, I also think 5.7 also makes a lot of sense for this specific PR, as you've mentioned...)

Reply:
Thanks for the writeup! That makes a lot of sense. I'm happy to follow this versioning scheme, knowing what it means and the intent behind it.

Reply:
Thank you for pointing out the fuzzy versioning methodology in the first place! It helps a lot to have to put these things into words, rather than having them be these ephemeral, unspoken thing. :)